lib/code_corps_web/controllers/user_controller.ex

```diff
@@ -1,40 +1,42 @@
 defmodule CodeCorpsWeb.UserController do
   use CodeCorpsWeb, :controller
-  use JaResource
-
-  import CodeCorps.Helpers.Query, only: [id_filter: 2, user_filter: 2, limit_filter: 2]
 
+  alias CodeCorps.{Helpers.Query, Services.UserService, User}
   alias CodeCorps.GitHub
-  alias CodeCorps.Services.UserService
-  alias CodeCorps.User
-
-  @spec model :: module
-  def model, do: CodeCorps.User
 
-  plug :load_and_authorize_resource, model: User, only: [:update]
-  plug JaResource
-  plug :login, only: [:create]
+  action_fallback CodeCorpsWeb.FallbackController
+  plug CodeCorpsWeb.Plug.DataToAttributes
 
-  def filter(_conn, query, "id", id_list) do
-    query |> id_filter(id_list)
+  @spec index(Conn.t, map) :: Conn.t
+  def index(%Conn{} = conn, %{} = params) do
+    with users <- User |> Query.id_filter(params) |> Query.limit_filter(params) |> Query.user_filter(params) |> Repo.all do
+      conn |> render("index.json-api", data: users)
+    end
   end
 
-  def handle_index(_conn, params) do
-    User
-    |> user_filter(params)
-    |> limit_filter(params)
+  @spec show(Conn.t, map) :: Conn.t
+  def show(%Conn{} = conn, %{"id" => id}) do
+    with %User{} = user <- User |> Repo.get(id) do
+      conn |> render("show.json-api", data: user)
+    end
   end
 
-  def handle_create(_conn, attributes) do
-    %User{} |> User.registration_changeset(attributes)
+  @spec create(Conn.t, map) :: Conn.t
+  def create(%Conn{} = conn, %{} = params) do
+    with {:ok, %User{} = user} <- %User{} |> User.registration_changeset(params) |> Repo.insert
+    do
+      conn |> Plug.Conn.fetch_session |> Guardian.Plug.sign_in(user) |> put_status(:created) |> render("show.json-api", data: user)
```

Comment:
@begedin so this worked.

Reply:
Although a bit lost on why this is needed as compared to before these changes.

Reply:
The `login` function was a plug. It automatically ran before performing the `create` action and told the conn that, after performing `create`, it needs to run the `do_login` function.

The `do_login` function then signed the `user` into the `conn` if the `create` action was successful.

Basically, from a user's perspective, "i want to be signed in when I register".

We had to do it via a plug, due to some quirks with `canary`.

Now that `canary` is gone, we can do the more direct route, so we just do it as part of the action itself.

Reply:
If you mean the `fetch_session` part, that should not be needed. What happens if you remove it?

Reply:
@begedin <img width="556" alt="screen shot 2017-09-05 at 7 46 21 am" src="https://user-images.githubusercontent.com/7374640/30070281-d07f1c9c-9217-11e7-86e0-3fde22013d09.png">

Reply:
@snewcomer Strange, but it makes sense.

I think you actually found a bug in the old login function. We were just calling `Conn.assign` instead of actually formally signing the user in. 

Additionally, it looks like our client is not expecting to sign the user in automatically upon registration and instead does it manually.

Considering these two, I think the simplest thing we can do is outright remove both `fetch_session` and `sign_in`.

Simply put, the code should read `conn |> put_status(:created) |> render("show.json-api", data: user)`and we can leave it that that. 

Sorry to make go through all of this, but hey, we discovered a bug and a piece of code that does nothing along the way!